cumulus/pallets/xcmp-queue/src/lib.rs

```diff
@@ -209,142 +176,161 @@ pub mod pallet {
-
-			Ok(())
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn integrity_test() {
+			let w = Self::on_idle_weight();
+			assert!(w != Weight::zero());
+			assert!(w.all_lte(T::BlockWeights::get().max_block));
 		}
 
-		/// Overwrites the speed to which the available weight approaches the maximum weight.
-		/// A lower number results in a faster progression. A value of 1 makes the entire weight
-		/// available initially.
-		///
-		/// - `origin`: Must pass `Root`.
-		/// - `new`: Desired value for `QueueConfigData.weight_restrict_decay`.
-		#[pallet::call_index(7)]
-		#[pallet::weight((T::WeightInfo::set_config_with_weight(), DispatchClass::Operational,))]
-		pub fn update_weight_restrict_decay(origin: OriginFor<T>, new: Weight) -> DispatchResult {
-			ensure_root(origin)?;
-			QueueConfig::<T>::mutate(|data| data.weight_restrict_decay = new);
+		fn on_idle(_block: BlockNumberFor<T>, limit: Weight) -> Weight {
+			use migration::{v3, LOG};
+			let mut meter = WeightMeter::with_limit(limit);
 
-			Ok(())
-		}
+			if meter.try_consume(Self::on_idle_weight()).is_err() {
+				log::debug!(target: LOG, "Not enough weight for on_idle. {} < {}", Self::on_idle_weight(), limit);
+				return meter.consumed()
+			}
 
-		/// Overwrite the maximum amount of weight any individual message may consume.
-		/// Messages above this weight go into the overweight queue and may only be serviced
-		/// explicitly.
-		///
-		/// - `origin`: Must pass `Root`.
-		/// - `new`: Desired value for `QueueConfigData.xcmp_max_individual_weight`.
-		#[pallet::call_index(8)]
-		#[pallet::weight((T::WeightInfo::set_config_with_weight(), DispatchClass::Operational,))]
-		pub fn update_xcmp_max_individual_weight(
-			origin: OriginFor<T>,
-			new: Weight,
-		) -> DispatchResult {
-			ensure_root(origin)?;
-			QueueConfig::<T>::mutate(|data| data.xcmp_max_individual_weight = new);
+			let Some(mut states) = v3::InboundXcmpStatus::<T>::get() else {
+				log::debug!(target: LOG, "Lazy migration finished: item gone");
+				return meter.consumed();
+			};
+			let Some(ref mut next) = states.first_mut() else {
+				log::debug!(target: LOG, "Lazy migration finished: item empty");
+				v3::InboundXcmpStatus::<T>::kill();
+				return meter.consumed();
+			};
+			log::debug!(
+				"Migrating inbound HRMP channel with sibling {:?}, msgs left {}.",
+				next.sender,
+				next.message_metadata.len()
+			);
+			// We take the last element since the MQ is a FIFO and we want to keep the order.
+			let Some((block_number, format)) = next.message_metadata.pop() else {
+				states.remove(0);
+				v3::InboundXcmpStatus::<T>::put(states);
+				return meter.consumed();
+			};
+			if format != XcmpMessageFormat::ConcatenatedVersionedXcm {
+				log::warn!(target: LOG,
+					"Dropping message with format {:?} (not ConcatenatedVersionedXcm)",
+					format
+				);
+				InboundXcmpMessages::<T>::remove(&next.sender, &block_number);
```

Comment:
Could we use `v3::InboundXcmpMessages` here and remove `InboundXcmpMessages` completely ?